### ISSUE TITLE
`tuist generate` trying to access iOS device when resolving package dependencies using xcodebuild

### ISSUE THREAD

**Issue by peng90** [tuist/tuist#2747]
**Describe the bug**
`tuist generate` tries to access iOS device when resolving package dependencies using `xcodebuild`. `xcodebuild` shows device access related errors if the iOS device is not unlocked when plugged in. See attached txt for full logs.

[tuist_logs.txt](https://github.com/tuist/tuist/files/6225204/tuist_logs.txt)

**To Reproduce**

1. Clone [this repo](https://github.com/peng90/tuist-bug-demo) for a reproducible test case.
2. Plug in an iOS device and make sure you don't unlock it while doing it
3. cd into the test case folder and run `tuist generate`

**Expected behavior**
`tuist generate` shouldn't trigger `xcodebuild` device access errors?

**Desktop (please complete the following information):**

- OS: macOS
- Version: 10.15.7

**Additional context**
The error happens in both Xcode 12.4 and 12.3. It doesn't seem to have any impact on building and running the generated project.

**Comment by fortmarek** [comment]
Hey @peng90!

Thanks for the issue and adding an example project - unfortunately, I cannot reproduce the issue with the steps you provided 😞 

Under the hood, we call `xcodebuild -resolvePackageDependencies`, could you run this command for your generated project and see if the same error occurs? Maybe specifying the destination to `generic/platform=iOS` could fix it (if at all possible, please try that as well).

Let me know if any of that helped!

**Comment by peng90** [comment]
Hi @fortmarek, thanks for responding! 

It's a bit tricky to reproduce the issue but the "reliable" way I found is to plugin in a device that's never been used for development for the day (aka it's the first time the device has been plugged in for the day) and make sure you don't unlock it when you plug it in. (I do it with my phone facing down...)

I did try `xcodebuild -resolvePackageDependencies` like you suggested but go the same error. Adding destination did help solving it to some degree... 

I get this error (see attached txt) a couple of times. It seems to be a bug from Apple? I haven't been able to find a more reliable way to reproduce this other than just running `xcodebuild -resolvePackageDependencies -destination 'generic/platform=iOS'` repeatedly until it shows up...
[tuist_logs_2.txt](https://github.com/tuist/tuist/files/6246825/tuist_logs_2.txt)
